libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/HasKotlinDependencies.kt

```diff
@@ -64,6 +64,18 @@ interface KotlinDependencyHandler {
         message = "Use npm(name, version) instead. Name like in package.json"
     )
     fun npm(org: String? = null, packageName: String, version: String = "*"): Dependency
+
+    fun devNpm(name: String, version: String = "*"): Dependency
+
+    fun devNpm(name: String, directory: File): Dependency
+
+    fun devNpm(directory: File): Dependency
+
```

Comment:
Minor: if you want, you can cover `optionalDependencies`
Its semantic is similar with normal dependencies 
And JFYI, I started to solve this issue right before you send PR :) 
Now I paused it, but you can take a look, can be a bit useful
https://github.com/JetBrains/kotlin/tree/rr/gradle/ilgonmic/npm-scopes